src/components/CustomProperties/CustomProperties.tsx

```diff
@@ -29,28 +34,30 @@ export function CustomProperties(props: CustomPropertiesProps) {
     colorScheme = DEFAULT_COLOR_SCHEME,
   } = props;
 
-  useEffect(() => {
-    let styleSheet = document.getElementById(STYLE_SHEET_ID);
+  const [hasInjectedStyleSheet, setHasInjectStyleSheet] = useState(
+    canUseDOM ? Boolean(document.getElementById(STYLE_SHEET_ID)) : false,
+  );
```

Comment:
I would like a comment here explaining this. It's important and I don't want it removed unintentionally.

Reply:
Something like this?
```js
  // This flag is used to prevent a flash of unstyled content when the
  // component mounts. Since the main stylesheet is injected in the useEffect
  // we don't want to render children until they have access to the custom properties.
  const [hasInjectedStyleSheet, setHasInjectStyleSheet] = useState(
```

Reply:
https://github.com/Shopify/polaris-react/pull/5299/commits/3609e83604d76149d4f7130d7d6a3a474c3ac59a